kolibri_instant_schools_plugin/assets/src/views/reset-password-modal/index.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+
+  <core-modal
+    :title="currentTitle"
+    @cancel="closeModal"
+  >
+    <div class="contents">
+      <status
+        v-if="showStatus"
+        :status="status"
+        @close="handleClose"
+      />
+      <phone-number-form
+        v-else
+        @submit="submitTokenRequest"
+        @close="closeModal"
+        :disabled="disableForms"
+      />
+    </div>
+  </core-modal>
+
+</template>
+
+
+<script>
+
+  import coreModal from 'kolibri.coreVue.components.coreModal';
+  import status from './status';
+  import phoneNumberForm from './phone-number-form';
+  import { createResetToken } from '../../state/resetPasswordActions';
+  import { RequestTokenStates as STATES } from '../../constants';
+
+  export default {
+    name: 'resetPasswordModal',
+    components: {
+      coreModal,
+      phoneNumberForm,
+      status,
+    },
+    data() {
+      return {
+        disableForms: false,
+        status: STATES.ENTER_PHONE_NUMBER,
+      };
+    },
+    computed: {
+      currentTitle() {
+        switch (this.status) {
+          case STATES.ACCOUNT_NOT_FOUND:
+            return this.$tr('accountNotFound');
+          case STATES.MESSAGE_SENT:
+            return this.$tr('messageSent');
+          case STATES.SMS_SERVICE_ERROR:
+            return this.$tr('smsServiceError');
+          default:
+            return this.$tr('resetPassword');
+        }
+      },
+      showStatus() {
+        return this.status !== STATES.ENTER_PHONE_NUMBER;
+      },
+    },
+    methods: {
+      submitTokenRequest(phoneNumber) {
+        this.disableForms = true;
+        this.createResetToken({ phoneNumber })
+          .then(() => {
+            this.status = STATES.MESSAGE_SENT;
+          })
+          .catch(err => {
+            const { code } = err.status;
+            if (code === 400) {
+              this.status = STATES.ACCOUNT_NOT_FOUND;
+            } else if (code === 500) {
```

Comment:
Could there ever be a chance of a different error code?

Reply:
Not based on the implementation, unless Django will randomly send a 5xx error. I could make this a catch all instead of checking specifically for 500.